File: Exercises/Ex6/pinkaew/random_norm.cc
#include "TFile.h"
#include "TTree.h"
#include "TRandom3.h"

int main(){
    size_t N = 1000;
    UInt_t seed = 5679;
    TRandom3 *generator = new TRandom3(seed);

    TFile *f = TFile::Open("out.root", "RECREATE");
    f -> cd();

    TTree * t = new TTree("tree", "title");
    Double_t s;
    t -> Branch("normal", &s, "normal/D");

    for(size_t i = 0; i < N; i++){
        s = generator -> Gaus();
        t -> Fill();
    }

    t -> Write();
    f -> Close();

    return 0;
}

File: Exercises/Ex6/pinkaew/random_norm.py
#!/usr/bin/env python

import ROOT
from array import array

if __name__ == "__main__":
    N = 1000
    seed = 5679
    generator = ROOT.TRandom3(seed)

    f= ROOT.TFile.Open("out_py.root", "RECREATE")

    t = ROOT.TTree("tree", "title")
    s = array('d', [0])
    t.Branch("normal", s, "normal/D")

    for _ in range(N):
        s[0] = generator.Gaus()
        t.Fill()

    f.WriteObject(t, "tree")
    f.Close()